### PR TITLE
dialects: (csl) print dsd ops

### PR DESCRIPTION
**Pull request by n-io**
Enabling printing DSD ops to CSL.

### CODE REVIEW

File: xdsl/backend/csl/print_csl.py [hunk 1]
```diff
@@ -459,6 +463,75 @@ def print_block(self, body: Block):
...
+                    self.print("});")
+                case csl.GetFabDsdOp(
+                    sizes=extent,
+                    fabric_color=fabric_color,
+                    control=control,
+                    wavelet_index_offset=wavelet_index_offset,
+                    result=result,
+                ):
+                    self.print(
+                        f"{self._var_use(result)} = @get_dsd({self.attribute_value_to_str(result.type)}, .{{ "
```
[the remark this answers is not in the document]
Comment (n-io) [reply xdsl/backend/csl/print_csl.py]: changed to use `mlir_type_to_csl_type`

File: xdsl/backend/csl/print_csl.py [hunk 2]
```diff
@@ -459,6 +463,75 @@ def print_block(self, body: Block):
...
+                    ind_vars_str = ", ".join(ind_vars)
+                    accesses = [
+                        (f"{str(strides.data[i].value.data)} * " if strides else "")
+                        + ind_vars[i]
+                        + (f" + {str(offsets.data[i].value.data)}" if offsets else "")
+                        for i in range(len(ind_vars))
+                    ]
+                    accesses_str = ", ".join(accesses)
+                    self.print(
+                        f"{self._var_use(result)} = @get_dsd( {self.attribute_value_to_str(result.type)} .{{"
```
[the remark this answers is not in the document]
Comment (n-io) [reply xdsl/backend/csl/print_csl.py]: changed to use `mlir_type_to_csl_type`

File: xdsl/backend/csl/print_csl.py [hunk 3]
```diff
@@ -271,6 +273,8 @@ def attribute_value_to_str(self, attr: Attribute) -> str:
                 return str(val.data)
             case StringAttr() as s:
                 return f'"{s.data}"'
+            case csl.DsdType() as dsd:
+                return dsd.data
```
[the remark this answers is not in the document]
Comment (n-io) [reply xdsl/backend/csl/print_csl.py]: Removed these lines and used `mlir_type_to_csl_type` as suggested

File: xdsl/dialects/csl.py [hunk 4]
```diff
@@ -772,9 +772,9 @@ class GetFabDsdOp(_GetDsdOp):
     """
 
     name = "csl.get_fab_dsd"
-    fabric_color = opt_prop_def(ColorIdAttr)
+    fabric_color = operand_def(ColorType)
```
[the remark this answers is not in the document]
Comment (n-io) [reply xdsl/dialects/csl.py:775]: What we need here is the result of `@get_color()`. It's a compile-time known value, but known at their compile time not at our compile time, I suppose.

[the remark this answers is not in the document]
Comment (dk949) [reply xdsl/dialects/csl.py:775]: We could print a call to `get_color` like is done with [data task IDs](https://github.com/xdslproject/xdsl/blob/8804e9262fdf1b1d74fc6e091360ccde7f828c8f/xdsl/backend/csl/print_csl.py#L87) and keep using the attribute/property.

**Comment by AntonLydike** [comment]
I feel like reducing the number of ops tested in printing would make this PR actually more readable. Maybe it is enough to test ~10-20 of the ops covered by `case csl.BuiltinDsdOp(ops=ops):`?

**Comment by n-io** [comment]
> I feel like reducing the number of ops tested in printing would make this PR actually more readable. Maybe it is enough to test ~10-20 of the ops covered by `case csl.BuiltinDsdOp(ops=ops):`?

Fair point, would it make sense to test the printing of one of each? If not, we can take out more?

File: xdsl/dialects/csl.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (n-io) [reply xdsl/dialects/csl.py:775]: How certain are we that this is correct?

[the remark this answers is not in the document]
Comment (n-io) [reply xdsl/dialects/csl.py]: changed to properties as discussed offline, and added `.input_queue` and `.output_queue` fields

**Review by AntonLydike — APPROVED** [review]
Really nice work!

**Pull request merged**